### ISSUE TITLE
More features

### ISSUE THREAD

**Issue by tmntmntmn** [aelsabbahy/goss#83]
Hi
I think goss is a great tool . 
I tried to use instead of ServerSpec and Testinfra,
But, I felt the lack of some functions when actually used .

I want to add some of the features :
  verify user's shell and password .
  verify kernel parameters .
  verify the NIC
  verify package architecture.Currently I cannot verify RHEL and CentOS packages.
  Example:
  postgresql-ｘｘ.el6_6.i686
  postgresql-ｘｘ.el6_6.x86_64

**Comment by tmntmntmn** [comment]
I've changed package name options, package problem has been resolved .
Example:
rpm -q --queryformat "%{NAME}.%{ARCH}\n" postgresql
goss a package postgresql.i686
goss a package psotgresql.x86_64

However , goss aa service it does not work well .
Hmmm...

**Comment by aelsabbahy** [comment]
Thanks for the feedback! I have some follow up questions:
1) When you say "I want to add some of the features" do you mean you'll submit PRs on this or are you asking for these features to be added?

2) Regarding the features:
* user shell and password, this should be fairly straight forward to implement. https://github.com/opencontainers/runc/blob/master/libcontainer/user/user.go looks like it handles both cases, so just need to expose it.

* kernel parameters, I'm on the fence on this one, it's easy to do using command, but it's also a comment test so might benefit from a dedicated resource..
How to do it with command resource:
```yaml
command:
  sysctl -q -n vm.swappiness:
    exit-status: 0
    stdout:
    - "60"
```

* Verify the NIC, can you expand more on what you mean by this?

What is the issue you're having with `goss aa service`?
* What OS/version are you using?
* Results are you seeing?
* Results are you expecting?

**Comment by tmntmntmn** [comment]
Thank you for your reply.

As to say of you that I want to do, can almost achieved via Command.
But I do not want as much as possible Command uses...

>Verify the NIC, can you expand more on what you mean by this?

I want to verify that
network interface exists.
network interface ip address correct(ip addr show/ifconfig).

>What OS/version are you using?

RHEL/CentOS 6.5-6.7 7.x

Backgroud:
I am trying to automate the testing of about 1000 nodes of EC2 instance on aws(with Ansible).

Now, I'm using Testinfra. 
Because ServerSpec is difficult for the members who can not use ruby.
Testinfra with a simple python code runs in the simple configuration file.

But I want as much as possible to eliminate the code and want to simplify the operation.
ServerSpec and Testinfra operation on RHEL is difficult for the members who have no experience of ruby or python.

However goss is great to work only in the simple configuration file with very easy operation.
With goss, I feel that it is possible a variety with a simple operation than ServerSpec and Testinfra.
So, I need a little more function if possible.

**Comment by aelsabbahy** [comment]
I think what would help me understand the requirements better would be if you send me Testinfra example code for each of the test scenario you'd like goss to be able to handle.

Once you do that, I'll close this issue and open an issue for each one of those use-cases as a feature request if I feel they're in high-demand, for that ones I'm hesitant about, we can discuss further here.

Since you mentioned that you're using Ansible, you might be interested in https://github.com/indusbox/goss-ansible

**Comment by tmntmntmn** [comment]
Sorry for late reply. And, thank you.
These are the code of the function is not in the goss.
-config.py
```
# read from goss-like.yml & os.getenv()
# its returns list value

# kernel params name,value
kernel = [
    ('net.ipv4.ip_forward', 0),
    ('net.ipv4.conf.all.rp_filter', 2),
    ('net.ipv4.conf.default.rp_filter', 2),
]
# network interface value for aws ec2 interface,value
nic = [
    ('eth0', ['xx.xx.xx.xx'])
    ('eth1', ['xx.xx.xx.xx'])
]
# file system mountpoint,filesystem type,device,option value
mount =[
    ('/proc','proc','proc',['rw', 'relatime']),
    ('/','ext4','/dev/xvda1',['rw']),
]
```
-test_ec2_nics.py
```
# -*- coding: utf-8 -*-
import pytest
import config

@pytest.mark.parametrize('name,addresses', config.nic)

# for aws
def test_nics(Interface,name,addresses):
    assert Interface(name).exists == True
    for adress in addresses:
        assert adress in Interface(name).addresses
    #assert Interface(name).speed  == speed
```
-test_kerynel.py
```
# -*- coding: utf-8 -*-
import pytest
import config

@pytest.mark.parametrize('param,value', config.kernel)

def test_kernel_params(Sysctl,param,value):
    assert str(Sysctl(param)) == str(value)
```
-test_mounts.py
```
# -*- coding: utf-8 -*-
import pytest
import config

@pytest.mark.parametrize('name,filesystem,device,options', config.mount)

def test_mounts(MountPoint,name,filesystem,device,options):
    assert MountPoint(name).exists
    assert MountPoint(name).filesystem == filesystem
    assert MountPoint(name).device     == device
    for opt in options:
        assert opt in MountPoint(name).options
```

**Comment by aelsabbahy** [comment]
A lot of the features you've requested have been released in [v0.1.8](https://github.com/aelsabbahy/goss/releases/tag/v0.1.8)

User password check is the only one that hasn't been added yet. You can track that one here: #92
